guide/creating-your-bot/command-handling.md

```diff
@@ -83,110 +97,258 @@ discord-bot/
 
+:::: code-group
+::: code-group-item commands/ping.js
 ```js
 const { SlashCommandBuilder } = require('@discordjs/builders');
 
 module.exports = {
 	data: new SlashCommandBuilder()
 		.setName('ping')
-		.setDescription('Replies with Pong!'),
-	async execute(interaction) {
-		await interaction.reply('Pong!');
-	},
+		.setDescription('Replies with pong!'),
 };
 ```
+:::
+::::
 
-You can go ahead and do the same for the rest of your commands, putting their respective blocks of code inside the `execute()` function.
+Go ahead and do the same for `commands/server.js` and `commands/user.js`.
 
 ::: tip
-[`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports) is how you export data in Node.js so that you can [`require()`](https://nodejs.org/api/modules.html#modules_require_id) it in other files.
+[`module.exports`](https://nodejs.org/api/modules.html#modules_module_exports) is how you export data in Node.js so that you can [`require()`](https://nodejs.org/api/modules.html#modules_require_id) (import) it in other files.
 
 If you need to access your client instance from inside a command file, you can access it via `interaction.client`. If you need to access external files, packages, etc., you should `require()` them at the top of the file.
 :::
 
-## Reading command files
+Now that each command is in its own file, we can implement the command handler for our deployment script. (Notice that we got rid of the `SlashCommandBuilder` import here, because we don't need it in the deployment script anymore)
 
-In your `index.js` file, make these additions:
+:::: code-group
+::: code-group-item deploy-commands.js
+```js {4,6-10}
+const { REST } = require('@discordjs/rest');
+const { Routes } = require('discord-api-types/v9');
+const { clientId, guildId, token } = require('./config.json');
+const { readdirSync } = require('fs')
 
-```js {1-2,7}
-const fs = require('fs');
-const { Client, Collection, Intents } = require('discord.js');
-const { token } = require('./config.json');
+const commands = [];
+for (const file of readdirSync('./commands').filter(f => f.endsWith('.js'))) {
+	const command = require(`./commands/${file}`);
+	commands.push(command.data.toJSON());
+}
 
-const client = new Client({ intents: [Intents.FLAGS.GUILDS] });
+const rest = new REST({ version: '9' }).setToken(token);
 
-client.commands = new Collection();
+(async () => {
+	try {
+		await rest.put(
+			Routes.applicationGuildCommands(clientId, guildId),
+			{ body: commands },
+		);
+
+		console.log('Successfully registered application commands.');
+	} catch (error) {
+		console.error(error);
+	}
+})();
 ```
+:::
+::::
 
-::: tip
-[`fs`](https://nodejs.org/api/fs.html) is Node's native file system module. <DocsLink section="collection" path="class/Collection" /> is a class that extends JavaScript's native [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) class, and includes more extensive, useful functionality.
```

Comment:
Why was this tip removed?

Reply:
The tip about `fs` seemed unnecessary, given we already have an explanation for `fs.readdirSync()`.

The tip about `Collection` was moved further down (in the section for "command handler for `index.js`") in line 280.